### PR TITLE
Circuit value conversions

### PR DESCRIPTION
**Pull request by sinui0**
This PR updates the `mpc-circuits` interface to use native types instead of forcing the user to deal with bitvecs and corresponding conversions. This will reduce boilerplate, junk and errors in higher layer crates.

I also have created a circuit specification format which obsoletes bristol-fashion. It allows us to map types to circuit io and include metadata. The format for gates is the same for now, so we just need to copy paste the BF circuits into these specs until we come up with a circuit builder DSL.

### CODE REVIEW

**Review by themighty1 — APPROVED** [review]
thanks, LGTM

**Comment by th4s** [comment]
LGTM! This API for doing the input/output encoding/decoding will make it a lot easier to deal with circuits.

**Review by th4s — COMMENTED** [review]
LGTM! This API for doing the input/output encoding/decoding will make it a lot easier to deal with circuits.

File: mpc-circuits/src/circuit.rs [hunk 1]
```diff
@@ -541,44 +604,46 @@ impl Circuit {
...
+                duplicate_output_wire_ids
+            )));
         }
 
         let wire_ids: HashSet<usize> = HashSet::from_iter(wire_ids);
         let input_wire_ids: HashSet<usize> = HashSet::from_iter(input_wires.iter().copied());
         let gate_output_wire_ids: HashSet<usize> = HashSet::from_iter(gate_output_wire_ids);
-        let output_wire_ids: HashSet<usize> = HashSet::from_iter(output_wires.iter().copied());
-
-        // Check that all gate output wires in last layer are assigned ot an Output
```
[the remark this answers is not in the document]
Comment (th4s) [reply mpc-circuits/src/circuit.rs:559]: oh my... started using a new difftool :see_no_evil:

**Pull request merged**